influxql/ast.go

```diff
@@ -642,6 +642,35 @@ type SelectStatement struct {
 	FillValue interface{}
 }
 
+// HasDerivative returns true if one of the fields in the statement is a
+// derivative aggregate
+func (s *SelectStatement) HasDerivative() bool {
+	for _, f := range s.FunctionCalls() {
+		if strings.HasSuffix(f.Name, "derivative") {
```

Comment:
A suggestion, not a blocker right now, but you could instead add `isDerivative()` to functions.go. I added `IsNumeric()` recently. That way all the code for checking stuff about functions is in one file. Then this code becomes:

```
if f.IsDerivative() {
...
}
```

It would also allow the check to be more precise -- if "derivate" or "non_negative_derivative".